guide.md

```diff
@@ -0,0 +1,440 @@
+# A guide to rules_nixpkgs
+
+TODO: is the flake shell pure? because if this is run on a non-nixos system, and /usr/bin/gcc continues to be visible, that messes things up
+TODO: document nixpkgs.nix
+
+---
+
+Bazel is a modern build system valued for performant, correct, and hermetic
+builds. It has a few shortcomings, one of which is that that Bazel is only
+hermetic if certain conditions are met, and they aren't in many people's build
```

Comment:
```suggestion
Bazel is a modern build system valued for fast and correct builds.
Fast, thanks to advanced caching and distributed builds.
Correct, in that build artifacts always reflect the state of their declared inputs. In other words, you should never have to run `bazel clean`.
Bazel's caching and distributed build features depend on correct build definitions.

Bazel build actions run in isolation to ensure that their build definitions are correct and capture all requirements.
This is referred to as "hermeticity": A build is hermetic if there are no external influences on it. However, Bazel is only
hermetic if certain conditions are met, and they aren't in many people's build
```
This uses some Bazel jargon, like "hermetic", without explanation. Maybe it could be expanded a bit to be more approachable to people not embedded in that jargon, yet.

Reply:
Good point, done (and similarly elsewhere throughout the document)